Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -734,31 +748,30 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
     return (NSInteger)[self.threadMappings numberOfSections];
 }
 
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)aSection
 {
-    NSInteger result = (NSInteger)[self.threadMappings numberOfItemsInSection:(NSUInteger)section];
-    if (self.hasArchivedThreadsRow) {
-        // Add the "archived conversations" row.
-        result++;
```

Comment:
Rather than offset indexes in a section managed by a YapDB mapping, I've added a new (potentially empty) section. This let's the yap logic work untouched in that section, and we can manually handle the other non-content sections (reminder views, archive button), which I think is a bit more flexible / less confusing.